extensions/amp-story/1.0/amp-story.js

```diff
@@ -1174,6 +1183,30 @@ export class AmpStory extends AMP.BaseElement {
     });
   }
 
+  /**
+   * Update the story level stack and check for navigation adherence
+   * to the path a user takes.
+   * @param {string} targetPageId
+   * @param {string} direction
+   * @private
+   */
+  updateStoryPath_(targetPageId, direction) {
+    const targetPage = this.getPageById(targetPageId);
+    if (direction == 'previous') {
+      this.storyPath_.pop();
```

Comment:
curious: have you tried navigating to a page in the story, navigating outside of the story, clicking back in the browser history to re-enter the story, and then navigating to the previous page? Will that still work?

Reply:
Thanks for bringing this case to attention, @Enriqe . You're right, the expected previous navigation is broken when the user navigates out of and comes back to the story. 

I will send a follow-up PR that updates the browser history like @gmajoulet suggested here and offline.